numba/typed/typedlist.py

```diff
@@ -317,44 +321,52 @@ def __gt__(self, other):
     def __ge__(self, other):
         return _ge(self, other)
 
-    def append(self, item):
+    def append(self, item: T) -> None:
         if not self._typed:
             self._initialise_list(item)
         _append(self, item)
 
-    def __setitem__(self, i, item):
+    def __setitem__(self, i, item: T) -> None:
         if not self._typed:
             self._initialise_list(item)
         _setitem(self, i, item)
 
-    def __getitem__(self, i):
+    # noqa F811 comments required due to github.com/PyCQA/pyflakes/issues/592
+    # noqa E704 required to follow overload style of using ... in the same line
+    @pt.overload
+    def __getitem__(self, i: int) -> T: ... # noqa: F811, E704
+    @pt.overload
+    def __getitem__(self, i: slice) -> 'List': ... # noqa: F811, E704
```

Comment:
Should this be `List[T]` as it is here? https://github.com/python/typeshed/blob/master/stdlib/3/builtins.pyi#L720

Reply:
yes, somehow I hadn't thought of it when writing it as text, but it makes sense.